service/raft/raft_group0.cc

```diff
@@ -298,6 +301,9 @@ future<> raft_group0::start_server_for_group0(raft::group_id group0_id) {
 
     group0_log.info("Server {} is starting group 0 with id {}", my_addr.id, group0_id);
     co_await _raft_gr.start_server_for_group(create_server_for_group0(group0_id, my_addr));
+    if (_status_for_monitoring == status_for_monitoring::not_started) {
```

Comment:
Why do you need to check previous state?

Reply:
`on_background_error` could be triggered before we leave `start_server_for_group`

Reply:
OK. I wonder if we need the `not_started` state at all. Look like it just complicates things without providing much benefit.

Reply:
What is the initial status otherwise? `disabled`? It's confusing, what if raft takes a while to start, the user won't be able to tell the difference from when the raft is really disabled.

Reply:
Initial status is `normal`. The purpose of the status is not let user monitor each state transition, but to detect an error state.

Reply:
Agree, will drop it.

Reply:
dropped